### DISCUSSION TITLE
How to disable writes to all repos in an organisation in Github Enterprise Server 2.10?

### DISCUSSION THREAD

**Discussion by sukumarp2022** [community/community#46583] [Enterprise]
### Select Topic Area

Question

### Body

I'm in the process of migrating hundreds of repos from GHES to GHEC.

During migration the users aren't supposed to commit anything or push anything to any repo in the organisation.

Is there any way to restrict the pushes all the repos in an organisation in GHES 2.10?

**Comment by 2MuchC0ff33** [comment]
Thanks for your question, @sukumarp2022 

Branch protections can be used to prevent writes to all repositories in a GitHub Enterprise Server 2.10 organisation. Branch protections enable you to impose specific rules on your branches, such as requiring pull request reviews before merging or passing status checks before merging.

Branch protections can be enforced by specifying the required status checks and pull request reviews in the organisation's default branch of each repository.

Here's how to make branch protections stick:

1. Go to the repository's settings
2. Select "Branches"
3. Click on the default branch (usually "main")
4. Scroll down to the "Protected branches" section
5. Under "Push restrictions", check the box next to "Protect this branch"
6. Under "Required status checks", select "Require status checks to pass before merging" and check all the checks you want to enforce
7. Under "Required pull request reviews", select "Require pull request reviews before merging"
8. Repeat these steps for each repository in the organisation.

Once these restrictions are in place, no one can push to the default branch unless the necessary conditions are met.

The following information is taken from the official GitHub Enterprise Server 2.10 documentation:
[[](https://docs.github.com/en/enterprise/2.10/user/github/administering-a-repository/enabling-branch-restrictions)](https://docs.github.com/en/enterprise/2.10/user/github/administering-a-repository/enabling-branch-restrictions)

**Reply by sukumarp2022** [reply]
The problem here is I have a number of repositories in the organisation. 
Manually updating branch protection rules for each and every repository is not feasible. 
Do we have any such restriction option at organisation level?

**Reply by 2MuchC0ff33** [reply]
No, sorry, @sukumarp2022, GitHub Enterprise Server (GHES) does not allow you to limit pushes at the organisation level. As you mentioned, you would need to set branch protection rules for each repository in your organisation individually. If you have many repositories, you could automate this process using the GitHub API or a scripting solution.
